packages/strapi-tiptap-editor/admin/src/utils/generateLabel.ts

```diff
@@ -0,0 +1,40 @@
+import truncate from 'lodash.truncate';
+import striptags from 'striptags';
+
+interface GenerateLabelData {
+  name: string;
+  content: string;
+}
+
+export const generateLabel = ({ name, content }: GenerateLabelData): { label: string; name: string } | null => {
+  if (!content?.trim()) return null;
+
+  // Preserve new lines by replacing block elements with \n before stripping tags
+  const formattedContent = content.replace(/<\/(p|div|h[1-6]|li|blockquote|tr|td)>/gi, '\n');
```

Comment:
Een label zou misschien geen newlines moeten hebben, sterker nog, het lijkt alsof je die weghaalt in de code erna.

```suggestion
  const formattedContent = content.replace(/<\/(p|div|h[1-6]|li|blockquote|tr|td)>/gi, ' ');
```